dataline-scheduler/src/test/java/io/dataline/scheduler/DefaultSchedulerPersistenceTest.java

```diff
@@ -0,0 +1,338 @@
+  private static PostgreSQLContainer container;
+  private static BasicDataSource connectionPool;
+
+  private static final SourceConnectionImplementation sourceConnectionImplementation;
+  private static final DestinationConnectionImplementation destinationConnectionImplementation;
+  private static final StandardSync standardSync;
+  private static final Instant now;
+
+  static {
+    final UUID workspaceId = UUID.randomUUID();
+    final UUID sourceImplementationId = UUID.randomUUID();
+    final UUID sourceSpecificationId = Integrations.POSTGRES_TAP.getSpecId();
+
+    JsonNode implementationJson = Jsons.jsonNode(ImmutableMap.builder()
+        .put("apiKey", "123-abc")
+        .put("hostname", "dataline.io")
+        .build());
+
+    sourceConnectionImplementation = new SourceConnectionImplementation();
+    sourceConnectionImplementation.setWorkspaceId(workspaceId);
+    sourceConnectionImplementation.setSourceSpecificationId(sourceSpecificationId);
+    sourceConnectionImplementation.setSourceImplementationId(sourceImplementationId);
+    sourceConnectionImplementation.setConfiguration(implementationJson);
+    sourceConnectionImplementation.setTombstone(false);
+
+    final UUID destinationImplementationId = UUID.randomUUID();
+    final UUID destinationSpecificationId = Integrations.POSTGRES_TARGET.getSpecId();
+
+    destinationConnectionImplementation = new DestinationConnectionImplementation();
+    destinationConnectionImplementation.setWorkspaceId(workspaceId);
+    destinationConnectionImplementation.setDestinationSpecificationId(destinationSpecificationId);
+    destinationConnectionImplementation.setDestinationImplementationId(destinationImplementationId);
+    destinationConnectionImplementation.setConfiguration(implementationJson);
+
+    final Column column = new Column();
+    column.setDataType(DataType.STRING);
+    column.setName("id");
+    column.setSelected(true);
+
+    final Table table = new Table();
+    table.setName("users");
+    table.setColumns(Lists.newArrayList(column));
+    table.setSelected(true);
+
+    final Schema schema = new Schema();
+    schema.setTables(Lists.newArrayList(table));
+
+    final UUID connectionId = UUID.randomUUID();
+
+    standardSync = new StandardSync();
+    standardSync.setConnectionId(connectionId);
+    standardSync.setName("presto to hudi");
+    standardSync.setStatus(StandardSync.Status.ACTIVE);
+    standardSync.setSchema(schema);
+    standardSync.setSourceImplementationId(sourceImplementationId);
+    standardSync.setDestinationImplementationId(UUID.randomUUID());
+    standardSync.setSyncMode(StandardSync.SyncMode.APPEND);
+
+    now = Instant.now();
+    @SuppressWarnings("unchecked")
+    Supplier<Instant> timeSupplier = mock(Supplier.class);
+    when(timeSupplier.get()).thenReturn(now);
+
+  }
+
+  @SuppressWarnings("rawtypes")
+  @BeforeAll
+  public static void dbSetup() {
+    container = new PostgreSQLContainer("postgres:13-alpine")
```

Comment:
@michel-tricot - i played with the idea of creating a base test case that sets this up (it's identical to some of the tests in dataline-db). i think from what you were saying this morning, you'd prefer to go with the way it's done here and just eat the duplication. is that right?